### ISSUE TITLE
Improve access control for markdown

### ISSUE THREAD

**Issue by danieladugyan** [Dsek-LTH/web#440]
Right now, `markdowns:update` is needed to be able to edit any content on the page. This is a regression from how it used to previously work: we had read/write permission for every markdown item, i.e `markdowns:sexm:read`. Restore the missing functionality.

**Comment by Macludde** [comment]
No, I don't think so. In zmodel markdowns:update is required but when you are updating a specific markdown, we check if you have markdowns:sexm:update and then use authorizedPrismaClient instead.

This is due to a limitation of the zenstack zmodel language, where we cannot check if a user's policies include markdowns:{name}:update (it doesn't support string interpolation, it doesn't support loops or a few other things). So only way to do it I found is to handle access manually for this

**Comment by jiashengguo** [comment]
Hey, although I'm not quite sure about the problem, ZenStack has recently added a `check` API which seems related:

https://zenstack.dev/docs/guides/check-permission

If that's not what you need, would you mind creating a FR in ZenStack's repo?
